builtin/logical/database/credentials.go

```diff
@@ -170,3 +175,191 @@ func (kg rsaKeyGenerator) configMap() (map[string]interface{}, error) {
 	}
 	return config, nil
 }
+
+type ClientCertificateGenerator struct {
```

Comment:
We can create a separate task for this, but it would be good to add comments to each of the exported fields here to indicate if they're required/optional and what values can be provided to them (similar to what's done in `rsaKeyGenerator`).

Reply:
done in [0abd0a4](https://github.com/hashicorp/vault/pull/20425/commits/0abd0a450a0cd0e2e7723659150ab349a0061002)